src/InverseProblems.jl

```diff
@@ -207,9 +216,12 @@ function transform_time_series(output_map::ConcatenatedOutputMap, time_series::S
 
     for field_name in keys(time_series.field_time_serieses)
         field_time_series = time_series.field_time_serieses[field_name]
+        A = Array(interior(field_time_series))
 
-        # Copy time series data to `Array`, keeping only `time_indices` and discarding halos
-        field_time_series_data = Array(interior(field_time_series))[:, :, :, output_map.time_indices]
+        # Ignore initial condition given by first element in map.time_indices
+        field_time_series_data = output_map.time_indices == Colon() ? 
+            selectdim(A, 4, 2:size(A,4)) :
+            selectdim(A, 4, output_map.time_indices[2:end])
```

Comment:
Does this work on the GPU? I think we copy and transfer to GPU intentionally above (by passing the result through `Array`). But it looks like `selectdim` uses `view`. Also we are not omitting halos from `field_time_series_data`?